index.js

```diff
@@ -117,22 +119,27 @@ module.exports = function (source) {
     // when this loader compiles multiple templates.
     compiledTemplate += 'var dependencies = nunjucks.webpackDependencies || (nunjucks.webpackDependencies = {});\n';
 
-    var templateReg = /env\.getTemplate\(\"(.*?)\"/g;
-    var match;
+    var templateReg = /(?<=env\.getTemplate\(")(.+?)(?=")/g;    // Node >= 8
+    var isRelativeReg = /^\.\.?\//;
 
     // Create an object to store references to the dependencies that have been included - this ensures that a template
     // dependency is only written once per file, even if it is used multiple times.
     var required = {};
 
     // Iterate over the template dependencies
-    while (match = templateReg.exec(nunjucksCompiledStr)) {
-        var templateRef = match[1];
-        if (!required[templateRef]) {
-            // Require the dependency by name, so it gets bundled by webpack
-            compiledTemplate += 'dependencies["' + templateRef + '"] = require( "' + templateRef + '" );\n';
-            required[templateRef] = true;
+    nunjucksCompiledStr = nunjucksCompiledStr.replace(templateReg, function(templateRef) {
```

Comment:
What was the reason to replace `while {}` with `.replace()`?

Reply:
This is to replace paths relative to the current template with paths relative to the root dir (e.g. so `./default.njk` becomes `rootDir/pathTo/default.njk`) in the `nunjucksCompiledStr` itself so they match keys in the `dependencies` object.

Reply:
Yep, but why don't do the same in `while {}`?

Reply:
`replace` seemed more more straightforward. It is possible with `while`, but looks awkward:
```javascript
while (match = templateReg.exec(nunjucksCompiledStr)) {
	// ...
	nunjucksCompiledStr = (
		nunjucksCompiledStr.slice(0, match.index)
		+ templateRefRel
		+ nunjucksCompiledStr.slice(match.index + templateRef.length)
	);
}
```